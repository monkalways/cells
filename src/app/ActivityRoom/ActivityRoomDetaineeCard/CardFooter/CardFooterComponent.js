import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Button, CardActions, Typography, withStyles,
} from '@material-ui/core';
import { withRouter } from 'react-router-dom';
import { compose } from 'recompose';

import CheckInDetaineeDialog from './CheckInDetaineeDialog';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({
    id: PropTypes.string.isRequired,
    activityRoomName: PropTypes.string.isRequired,
    location: PropTypes.string.isRequired,
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  isCheckingIn: PropTypes.bool.isRequired,
  isCheckingInSuccess: PropTypes.bool.isRequired,
  usage: PropTypes.string.isRequired,
  onCheckIn: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
  checkingInDetaineeId: PropTypes.string,
};

const defaultProps = {
  checkingInDetaineeId: null,
};

const LOCATION_IN_TRANSIT = 'in transit';
const LOCATION_IN_PROGRESS = 'in progress';

export class CardFooterComponent extends Component {
  state = {
    isDialogOpen: false,
  };

  componentWillReceiveProps(nextProps) {
    this.setState({
      isDialogOpen:
        nextProps.isCheckingIn
        && !nextProps.isCheckingInSuccess
        && nextProps.detainee.id === nextProps.checkingInDetaineeId,
    });
  }

  handleInButtonClick = () => {
    this.setState({
      isDialogOpen: true,
    });
  };

  handleOutButtonClick = () => {
    const { history, detainee, usage } = this.props;
    const first = 'activity-rooms';
    history.push(`/detainees/${detainee.id}?first=${first}&second=${usage}`);
  };

  handleClose = () => {
    this.setState({
      isDialogOpen: false,
    });
  };

  handleCheckIn = () => {
    const { onCheckIn, detainee, usage } = this.props;
    onCheckIn(detainee.id, usage);
  };

  renderButton = () => {
    const { detainee, classes } = this.props;
    const { location } = detainee;
    if (location) {
      if (location.toLowerCase().includes(LOCATION_IN_TRANSIT)) {
        return (
          <Button
            id="inButton"
            variant="contained"
            size="large"
            className={classes.inButton}
            onClick={this.handleInButtonClick}
          >
            IN
          </Button>
        );
      }
      if (location.toLowerCase().includes(LOCATION_IN_PROGRESS)) {
        return (
          <Button
            id="outButton"
            variant="contained"
            size="large"
            className={classes.outButton}
            onClick={this.handleOutButtonClick}
          >
            OUT
          </Button>
        );
      }
    }
    return null;
  };

  render() {
    const {
      classes,
      isAuthenticated,
      detainee,
      isCheckingIn,
      usage,
    } = this.props;
    const { isDialogOpen } = this.state;
    return (
      <CardActions className={classes.actions}>
        {isAuthenticated && (
          <React.Fragment>
            <Typography variant="h6" className={classes.heading}>
              {detainee.activityRoomName}
            </Typography>
            {this.renderButton()}
            <CheckInDetaineeDialog
              isDialogOpen={isDialogOpen}
              detainee={detainee}
              isCheckingIn={isCheckingIn}
              usage={usage}
              onClose={this.handleClose}
              onCheckIn={this.handleCheckIn}
            />
          </React.Fragment>
        )}
      </CardActions>
    );
  }
}

CardFooterComponent.propTypes = propTypes;
CardFooterComponent.defaultProps = defaultProps;

export default compose(
  withStyles((theme) => ({
    actions: {
      display: 'flex',
      height: theme.spacing.unit * 9,
    },
    heading: {
      width: theme.spacing.unit * 8,
      marginRight: theme.spacing.unit * 4,
    },
    inButton: {
      backgroundColor: '#2895F2',
      color: '#FFFFFF',
      fontSize: theme.typography.h6.fontSize,
      '&:hover': {
        backgroundColor: '#0E47A1',
      },
    },
    outButton: {
      backgroundColor: '#558A3A',
      color: '#FFFFFF',
      fontSize: theme.typography.h6.fontSize,
      '&:hover': {
        backgroundColor: '#456336',
      },
    },
  })),
  withRouter,
)(CardFooterComponent);
