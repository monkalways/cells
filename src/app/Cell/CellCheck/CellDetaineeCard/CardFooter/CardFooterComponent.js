import React from 'react';
import PropTypes from 'prop-types';
import { CardActions, IconButton, withStyles } from '@material-ui/core';

import WellnessVisualIcon from '../../../../images/WellnessVisual.png';
import WellnessVerbalIcon from '../../../../images/WellnessVerbal.png';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  cellCheck: PropTypes.shape({
    visual: PropTypes.bool.isRequired,
    verbal: PropTypes.bool.isRequired,
  }),
  onVisualClick: PropTypes.func,
  onVerbalClick: PropTypes.func,
};

const defaultProps = {
  cellCheck: null,
  onVisualClick: null,
  onVerbalClick: null,
};

const CardFooterComponent = ({
  classes,
  cellCheck,
  onVisualClick,
  onVerbalClick,
}) => (
  <CardActions className={classes.actions} disableActionSpacing>
    {cellCheck && (
      <React.Fragment>
        <IconButton
          className={
            cellCheck.visual ? classes.leftButtonSelected : classes.leftButton
          }
          onClick={onVisualClick}
        >
          <img
            src={WellnessVisualIcon}
            alt="wellness visual"
            className={classes.img}
          />
        </IconButton>
        <IconButton
          className={
            cellCheck.verbal ? classes.rightButtonSelected : classes.rightButton
          }
          onClick={onVerbalClick}
        >
          <img
            src={WellnessVerbalIcon}
            alt="wellness verbal"
            className={classes.img}
          />
        </IconButton>
      </React.Fragment>
    )}
  </CardActions>
);

CardFooterComponent.propTypes = propTypes;
CardFooterComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  actions: {
    display: 'flex',
    height: theme.spacing.unit * 9,
  },
  leftButton: {
    margin: theme.spacing.unit,
    opacity: 0.2,
  },
  leftButtonSelected: {
    margin: theme.spacing.unit,
  },
  rightButton: {
    marginLeft: theme.spacing.unit * 5,
    margin: theme.spacing.unit,
    opacity: 0.2,
  },
  rightButtonSelected: {
    marginLeft: theme.spacing.unit * 5,
    margin: theme.spacing.unit,
  },
  img: {
    height: theme.spacing.unit * 6,
    width: theme.spacing.unit * 6,
  },
}))(CardFooterComponent);
