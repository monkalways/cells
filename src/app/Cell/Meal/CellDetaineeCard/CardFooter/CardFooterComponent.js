import React from 'react';
import PropTypes from 'prop-types';
import { CardActions, IconButton, withStyles } from '@material-ui/core';

import MealAcceptIcon from '../../../../images/MealAccept.png';
import MealDeclineIcon from '../../../../images/MealDecline.png';
import NotApplicableIcon from '../../../../images/NotApplicable.png';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  meal: PropTypes.shape({
    accept: PropTypes.bool.isRequired,
    reject: PropTypes.bool.isRequired,
    notApplicable: PropTypes.bool.isRequired,
  }),
  onAcceptClick: PropTypes.func.isRequired,
  onRejectClick: PropTypes.func.isRequired,
  onNotApplicableClick: PropTypes.func.isRequired,
};

const defaultProps = {
  meal: null,
};

const CardFooterComponent = ({
  classes,
  meal,
  onAcceptClick,
  onRejectClick,
  onNotApplicableClick,
}) => (
  <CardActions className={classes.actions} disableActionSpacing>
    {meal && (
      <React.Fragment>
        <IconButton
          className={meal.accept ? classes.buttonSelected : classes.button}
          onClick={onAcceptClick}
        >
          <img src={MealAcceptIcon} alt="meal accept" className={classes.img} />
        </IconButton>
        <IconButton
          className={meal.reject ? classes.buttonSelected : classes.button}
          onClick={onRejectClick}
        >
          <img
            src={MealDeclineIcon}
            alt="meal decline"
            className={classes.img}
          />
        </IconButton>
        <IconButton
          className={
            meal.notApplicable ? classes.buttonSelected : classes.button
          }
          onClick={onNotApplicableClick}
        >
          <img
            src={NotApplicableIcon}
            alt="not applicable"
            className={classes.img}
            disabled
          />
        </IconButton>
      </React.Fragment>
    )}
  </CardActions>
);

CardFooterComponent.propTypes = propTypes;
CardFooterComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  actions: {
    display: 'flex',
    height: theme.spacing.unit * 9,
  },
  button: {
    margin: 0,
    opacity: 0.2,
  },
  buttonSelected: {
    margin: 0,
  },
  img: {
    height: theme.spacing.unit * 6,
    width: theme.spacing.unit * 6,
  },
}))(CardFooterComponent);
