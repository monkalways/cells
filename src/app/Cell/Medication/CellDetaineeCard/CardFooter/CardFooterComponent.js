import React from 'react';
import PropTypes from 'prop-types';
import { CardActions, IconButton, withStyles } from '@material-ui/core';

import MedicineAcceptIcon from '../../../../images/MedicineAccept.png';
import MedicineDeclineIcon from '../../../../images/MedicineDecline.png';
import NotApplicableIcon from '../../../../images/NotApplicable.png';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  cellDetainee: PropTypes.shape({}).isRequired,
  medication: PropTypes.shape({
    accept: PropTypes.bool.isRequired,
    reject: PropTypes.bool.isRequired,
    notApplicable: PropTypes.bool.isRequired,
  }),
  onAcceptClick: PropTypes.func.isRequired,
  onRejectClick: PropTypes.func.isRequired,
  onNotApplicableClick: PropTypes.func.isRequired,
};

const defaultProps = {
  medication: null,
};

export const CardFooterComponent = ({
  classes,
  cellDetainee,
  medication,
  onAcceptClick,
  onRejectClick,
  onNotApplicableClick,
}) => (
  <CardActions className={classes.actions} disableActionSpacing>
    {!cellDetainee.location
      && cellDetainee.isUnderMedication
      && medication && (
        <React.Fragment>
          <IconButton
            id="acceptButton"
            className={
              medication.accept ? classes.buttonSelected : classes.button
            }
            onClick={onAcceptClick}
          >
            <img
              src={MedicineAcceptIcon}
              alt="Medicine accept"
              className={classes.img}
            />
          </IconButton>
          <IconButton
            id="rejectButton"
            className={
              medication.reject ? classes.buttonSelected : classes.button
            }
            onClick={onRejectClick}
          >
            <img
              src={MedicineDeclineIcon}
              alt="Medicine decline"
              className={classes.img}
            />
          </IconButton>
          <IconButton
            id="notApplicableButton"
            className={
              medication.notApplicable ? classes.buttonSelected : classes.button
            }
            onClick={onNotApplicableClick}
          >
            <img
              src={NotApplicableIcon}
              alt="not applicable"
              className={classes.img}
              disabled
            />
          </IconButton>
        </React.Fragment>
    )}
  </CardActions>
);

CardFooterComponent.propTypes = propTypes;
CardFooterComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  actions: {
    display: 'flex',
    height: theme.spacing.unit * 9,
  },
  button: {
    margin: 0,
    opacity: 0.2,
  },
  buttonSelected: {
    margin: 0,
  },
  img: {
    height: theme.spacing.unit * 6,
    width: theme.spacing.unit * 6,
  },
}))(CardFooterComponent);
