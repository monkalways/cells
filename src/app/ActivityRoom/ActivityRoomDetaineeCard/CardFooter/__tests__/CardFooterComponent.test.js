import React from 'react';
import { shallow } from 'enzyme';
import { Button, Typography } from '@material-ui/core';

import { CardFooterComponent } from '../CardFooterComponent';

describe('CardFooterComponent', () => {
  let isAuthenticated;
  let detaineeId;
  let location;
  let onCheckIn;
  let isCheckingIn;
  let isCheckingInSuccess;
  let push;
  let usage;

  beforeEach(() => {
    isAuthenticated = true;
    detaineeId = '123';
    location = 'Phone - In Transit';
    onCheckIn = jest.fn();
    isCheckingIn = false;
    isCheckingInSuccess = false;
    push = jest.fn();
    usage = 'phone';
  });

  const setup = () => {
    const classes = {
      actions: '',
      heading: '',
      inButton: '',
      outButton: '',
    };
    const detainee = {
      id: detaineeId,
      activityRoomName: 'T1',
      location,
      firstName: 'John',
      lastName: 'Smith',
    };
    const history = { push };

    return shallow(<CardFooterComponent
      classes={classes}
      history={history}
      detainee={detainee}
      isAuthenticated={isAuthenticated}
      isCheckingIn={isCheckingIn}
      isCheckingInSuccess={isCheckingInSuccess}
      onCheckIn={onCheckIn}
      usage={usage}
    />);
  };

  it('should set isDialogOpen to true when check-in is still in progress', () => {
    const wrapper = setup();
    const nextProps = {
      isCheckingIn: true,
      isCheckingInSuccess: false,
    };

    wrapper.instance().componentWillReceiveProps(nextProps);

    expect(wrapper).toHaveState('isDialogOpen', true);
  });

  it('should set isDialogOpen to false when check-in is complete', () => {
    const wrapper = setup();
    const nextProps = {
      isCheckingIn: false,
      isCheckingInSuccess: true,
    };

    wrapper.instance().componentWillReceiveProps(nextProps);

    expect(wrapper).toHaveState('isDialogOpen', false);
  });

  it('should render empty when isAuthenticated is false', () => {
    isAuthenticated = false;
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Typography)).not.toExist();
  });

  it('should render title when isAuthenticated is true', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Typography)).toExist();
  });

  it('should render IN Button when location contains "in transit"', () => {
    location = 'Phone - In Transit';
    const wrapper = setup();

    const inButton = wrapper.find('#inButton');
    expect(inButton).toExist();
    expect(wrapper).toHaveState('isDialogOpen', false);

    inButton.simulate('click');
    expect(wrapper).toHaveState('isDialogOpen', true);
  });

  it('should render OUT Button when location contains "in progress"', () => {
    location = 'Phone - In Progress';
    const wrapper = setup();

    const outButton = wrapper.find('#outButton');
    expect(outButton).toExist();

    outButton.simulate('click');
    expect(push).toBeCalledWith(`/detainees/${detaineeId}?first=activity-rooms&second=${usage}`);
  });

  it('should not render any Button when location does not contain "in transit" or "in progress"', () => {
    location = '';
    const wrapper = setup();

    expect(wrapper.find(Button)).not.toExist();
  });

  it('should handle close', () => {
    location = '';
    const wrapper = setup();

    wrapper.instance().handleClose();
    expect(wrapper).toHaveState('isDialogOpen', false);
  });

  it('should handle checkin', () => {
    location = '';
    const wrapper = setup();

    wrapper.instance().handleCheckIn();
    expect(onCheckIn).toBeCalledWith(detaineeId, usage);
  });
});
