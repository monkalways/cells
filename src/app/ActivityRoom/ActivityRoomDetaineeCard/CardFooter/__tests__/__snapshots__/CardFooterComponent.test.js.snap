// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooterComponent should render empty when isAuthenticated is false 1`] = `
<WithStyles(CardActions)
  className={Object {}}
/>
`;

exports[`CardFooterComponent should render title when isAuthenticated is true 1`] = `
<WithStyles(CardActions)
  className={Object {}}
>
  <WithStyles(Typography)
    className={Object {}}
    variant="h6"
  >
    T1
  </WithStyles(Typography)>
  <WithStyles(Button)
    className={Object {}}
    id="inButton"
    onClick={[Function]}
    size="large"
    variant="contained"
  >
    IN
  </WithStyles(Button)>
  <CheckInDetaineeDialogComponent
    detainee={
      Object {
        "activityRoomName": "T1",
        "id": "123",
        "location": "Phone - In Transit",
      }
    }
    isCheckingIn={false}
    isDialogOpen={false}
    onCheckIn={[Function]}
    onClose={[Function]}
    usage="phone"
  />
</WithStyles(CardActions)>
`;
